core/src/filed/restore.cc

```diff
@@ -398,6 +398,7 @@ void DoRestore(JobControlRecord* jcr)
   BareosSocket* sd;
   uint32_t VolSessionId, VolSessionTime;
   int32_t file_index;
+  int32_t prev_file_index = 0;
```

Comment:
The point is that file_index is never 0 (since FI are 1-counted) and so `prev_file_index` is always not equal to the first file index encountered.

Otherwise you would have to treat the case of the first file as special.